detect-child-process false-positive with spawn

```js
const { spawn } = require('child_process');
```

Gives the `Found require("child_process")` error.

According to [Avoiding Command Injection in Node.js](https://github.com/nodesecurity/eslint-plugin-security/blob/master/docs/avoid-command-injection-node.md), `child_process.spawn` is safer than `child_process.exec`.

So is it a false-positive, or does the `detect-child-process` rule tell me to completely avoid using `child_process`?

As described in the doc, `spawn` is secure: https://github.com/nodesecurity/eslint-plugin-security/blob/main/docs/avoid-command-injection-node.md#so-how-do-we-do-this-the-right-way

I think the confusion comes from the fact of having two same rule errors for a vulnerable `exec` example like this:

```js
const child_process = require('child_process');

var path = "user input";
child_process.exec('ls -l' + input, function (err, data) {
  console.log(data);
});
```

Output:
```
  1:23  warning  Found require("child_process")                              security/detect-child-process  
  4:1   warning  Found child_process.exec() with non Literal first argument  security/detect-child-process
```

Maybe we should include two different rules like: `detect-child-process` and `detect-child-process-exec`. As a developer or security researcher both data is interesting, but the `exec` related one is more important.

--- a/rules/detect-child-process.js
+++ b/rules/detect-child-process.js
@@ -47,12 +47,18 @@ module.exports = {
         if (node.callee.name === 'require') {
           const args = node.arguments[0];
           if (args && args.type === 'Literal' && args.value === 'child_process') {
+            let pattern;
             if (node.parent.type === 'VariableDeclarator') {
-              extractChildProcessIdentifiers(node.parent.id);
+              pattern = node.parent.id;
             } else if (node.parent.type === 'AssignmentExpression' && node.parent.operator === '=') {
-              extractChildProcessIdentifiers(node.parent.left);
+              pattern = node.parent.left;
             }
-            return context.report({ node: node, message: 'Found require("child_process")' });
+            if (pattern) {
+              extractChildProcessIdentifiers(pattern);
+            }
+            if (!pattern || pattern.type === 'Identifier') {
+              return context.report({ node: node, message: 'Found require("child_process")' });
+            }
           }
         }
       },

--- a/test/detect-child-process.js
+++ b/test/detect-child-process.js
@@ -7,7 +7,19 @@ const ruleName = 'detect-child-process';
 const rule = require(`../rules/${ruleName}`);
 
 tester.run(ruleName, rule, {
-  valid: ["child_process.exec('ls')"],
+  valid: [
+    "child_process.exec('ls')",
+    {
+      code: `
+      var {} = require('child_process');
+      var result = /hello/.exec(str);`,
+      parserOptions: { ecmaVersion: 6 },
+    },
+    {
+      code: "var { spawn } = require('child_process'); spawn(str);",
+      parserOptions: { ecmaVersion: 6 },
+    },
+  ],
   invalid: [
     {
       code: "require('child_process')",
@@ -24,13 +36,6 @@ tester.run(ruleName, rule, {
     {
       code: "var child = sinon.stub(require('child_process')); child.exec.returns({});",
       errors: [{ message: 'Found require("child_process")' }],
-    },
-    {
-      code: `
-      var {} = require('child_process');
-      var result = /hello/.exec(str);`,
-      parserOptions: { ecmaVersion: 6 },
-      errors: [{ message: 'Found require("child_process")', line: 2 }],
     },
     {
       code: `

